streams/shared/src/main/scala/zio/stream/experimental/ZTransducer.scala

```diff
@@ -124,6 +125,250 @@ object ZTransducer {
   def fromPush[R, E, I, O](push: Option[Chunk[I]] => ZIO[R, E, Chunk[O]]): ZTransducer[R, E, I, O] =
     ZTransducer(Managed.succeed(push))
 
+  /**
+   * Creates a transducer by folding over a structure of type `S`.
+   */
+  def fold[I, O](z: O)(contFn: O => Boolean)(f: (O, I) => (O, Chunk[I])): ZTransducer[Any, Nothing, I, O] =
```

Comment:
Why does the step function return `Chunk[I]` in addition to `O`? Is this for leftovers? If so, it isn't necessary; the operator can check the `contFn` after every application of `(O, I) => O` to see if it should continue applying `f` to the chunk.

Reply:
Nice, I blindly copied the existing one. There's another place where remainders point their nose, it's in `foldDecompose`. AFAICS, in the existing implementation, the result of `decompose(input)` is added to the leftovers, but since there is no such thing here, I have no other choice than to add it to the buffer. The danger is that decomposing is called ad nauseam.

Reply:
Perhaps we could borrow from the comonoid stuff and change decompose to `A => Option[(A, A)]`

Reply:
I guess the empty chunk can be None here.

Reply:
Meaning, if it returns the empty chunk, I have no other choice but to cross the maximum, otherwise I drop an element from the stream.